src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -949,3 +957,58 @@ export const handleUpdateForm: RequestHandler<
       return res.status(statusCode).json({ message: errorMessage })
     })
 }
+
+/**
+ * Handler for PATCH /form/:formId/settings.
+ * @security session
+ *
+ * @returns 200 with updated form settings
+ * @returns 400 when body is malformed; can happen when email parameter is passed for encrypt-mode forms
+ * @returns 403 when current user does not have permissions to update form settings
+ * @returns 404 when form to update settings for cannot be found
+ * @returns 409 when saving form settings incurs a conflict in the database
+ * @returns 410 when updating settings for archived form
+ * @returns 413 when updating settings causes form to be too large to be saved in the database
+ * @returns 422 when an invalid settings update is attempted on the form
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const handleUpdateSettings: RequestHandler<
+  { formId: string },
+  FormSettings | ErrorDto,
+  SettingsUpdateDto
+> = (req, res) => {
+  const { formId } = req.params
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+  const settingsToPatch = req.body
+
+  // Step 1: Retrieve currently logged in user.
+  return UserService.getPopulatedUserById(sessionUserId)
+    .andThen((user) =>
+      // Step 2: Retrieve form with write permission check.
+      AuthService.getFormAfterPermissionChecks({
+        user,
+        formId,
+        level: PermissionLevel.Write,
+      }),
+    )
+    .andThen((retrievedForm) =>
+      updateFormSettings(retrievedForm, settingsToPatch),
```

Comment:
I think the API design can be better modeled as an instance method instead i.e.  form.updateSettings(...)? Or perhaps even a static method in Form.updateSettings(...) that relies on Form.findOneAndUpdate() under the hood, so that there won't be concurrency issues from interleaved read-then-write operations.

Reply:
Good point, will make the change.

Reply:
Updated, but note that calling mongoose's `findByIdAndUpdate` requires the update to be in dot notation, or the same issue will occur where objects are overridden instead of merged